### DISCUSSION TITLE
Why doesn’t Pyright check that methods have a consistent type during multiple inheritance?

### DISCUSSION THREAD

**Discussion by Pyprohly** [microsoft/pyright#2270] [Q&A] [answered]
Pyright does not check that methods have a consistent type during multiple inheritance.

For instance, in the following:

```python
class A:
    def f(self) -> int:
        return 1

class B:
    def f(self) -> str:
        return 'b'

class C(A, B): pass
```

Mypy (0.910) says:

```
file.py:10: error: Definition of "f" in base class "A" is incompatible with definition in base class "B"
```

But Pyright (1.1.166) doesn’t complain.

**Comment by erictraut** [comment, marked as answer]
It's a good suggestion. I've added an enhancement request issue here: https://github.com/microsoft/pyright/issues/2273.

**Reply by erictraut** [reply]
This will be included in the next release of pyright.
